graphql_client/src/lib.rs

```diff
@@ -299,6 +299,12 @@ pub struct Response<Data> {
     pub extensions: Option<HashMap<String, serde_json::Value>>,
 }
 
+/// Hidden module for types used by the codegen crate.
+#[doc(hidden)]
+pub mod _private {
+    pub use ::serde;
+}
+
```

Comment:
I am conflicted on whether I like how this re-export works.
- First of all, I don't think you have to hide it. I think explicitly depending on serde 1 is not a problem. But I can't find best practices to back up that statement.
- Does this re-export belong here, or in `graphql_query_derive` as that crate defines the macro that uses it, or in `graphql_query_codegen`  as that crate produces the generated code?

Reply:
A hidden module is a common pattern for private types used in macros (see [serde](https://github.com/serde-rs/serde/blob/master/serde/src/lib.rs#L316), for an example). Adding a new public export is a bigger change than having it as a private module, regardless of whether publically exporting serde is an issue or not. I have not seen a crate publically re-export serde, though.

For the second point, this export needs to be in `graphql_client`. That is the only crate that is guaranteed to be available to users of the crate. Beyond that, you can't have non-proc-macro exports from a proc-macro crate which rules out exporting it from `graphql_query_derive`.

Reply:
> For the second point, this export needs to be in `graphql_client`. That is the only crate that we know will be available to users of the library. Beyond that, you can't have non-proc-macro exports from a proc-macro crate which rules out exporting it from `graphql_query_derive`.

I didn't know about that rule for proc-macro exports, then It makes total sense to do it in `graphql_client`.